pyfr/integrators/dual/phys/steppers.py

```diff
@@ -4,22 +4,127 @@
 
 
 class BaseDualStepper(BaseDualIntegrator):
-    pass
 
+    def _set_currstg(self, s):
```

Comment:
I am not a fan of this approach.  Are there any good alternatives which don't depend on manipulating internal state?

Reply:
I need to keep track of the current stage number somehow. Do you have any suggestions? Is it better to have this as a variable and set it directly without using such a function?

Reply:
Could we add some arguments to pseudo-advance, along with a function to get the stepper coeffs (which we can call once outside the main for loop since it is a function of dt which for this controller is fixed)?

Reply:
But stepper coeffs is a function of the stage number as well if we are using dirk steppers

Reply:
Yes, so I see us doing:

```
scoeffs = [self._stepper_coeffs(s, dt) for s in self.nstages]

while ...
   for sc in scoeffs:
      ...
```